cmd/routedns/main.go

```diff
@@ -192,9 +197,15 @@ func start(opt options, args []string) error {
 
 		switch l.Protocol {
 		case "tcp":
-			listeners = append(listeners, rdns.NewDNSListener(id, l.Address, "tcp", opt, resolver))
+			if portIsSet == false {
+				listenAddress = listenAddress + rdns.PlainDNSPort
+			}
+			listeners = append(listeners, rdns.NewDNSListener(id, listenAddress, "tcp", opt, resolver))
```

Comment:
You can simplify all this quite a bit. Just have a function that adds the port if it's not already on it. Something like 
```go
func addressWithDefault(addr, defaultPort string) string
```
That would let you change all instances to something like
```go
listenAddress = addressWithDefault(listenAddress, rdns.PlainDNSPort)
```
You could even do that inline if the lines don't get too long (probably better to keep it separate).

Another thing, when you're looking for `:` in an address, please use https://pkg.go.dev/net#SplitHostPort and https://pkg.go.dev/net#JoinHostPort. Then you can also remove the colon from the constants below.

Reply:
Done